site/server/app/com/fortysevendeg/exercises/models/UserModel.scala

```diff
@@ -1,100 +1,78 @@
+}
 
 trait UserStore {
+  def all: ConnectionIO[List[User]]
 
-  def all: Future[Seq[User]]
+  def getByLogin(login: String): ConnectionIO[Option[User]]
 
-  def getByLogin(login: String): Future[Seq[User]]
+  def getById(id: Long): ConnectionIO[Option[User]]
 
-  def create(
-    login:       String,
-    name:        String,
-    github_id:   String,
-    picture_url: String,
-    github_url:  String,
-    email:       String
-  ): Future[User]
+  def create(user: UserCreation.Request): ConnectionIO[UserCreation.Response]
 
-  def update(user: User): Future[Boolean]
+  def delete(id: Long): ConnectionIO[Boolean]
 
-  def delete(ids: Long*): Future[Boolean]
+  def update(user: User): ConnectionIO[Option[User]]
 }
 
-class UserSlickStore(db: Database) extends UserStore {
-
-  lazy val profile = current.configuration.getString("db.default.driver").collect {
-    case "org.h2.Driver"         ⇒ H2Driver
-    case "org.postgresql.Driver" ⇒ PostgresDriver
-    case "com.mysql.jdbc.Driver" ⇒ MySQLDriver
-    case other                   ⇒ throw new Exception(s"Unable to discern Slick driver for ${other}")
-  }.get
-
-  import play.api.db.DB
-  import profile.api._
-
-  class Users(tag: Tag) extends Table[User](tag, "users") {
-    def id = column[Option[Long]]("id", O.PrimaryKey, O.AutoInc)
-    def login = column[String]("login")
-    def name = column[String]("name")
-    def github_id = column[String]("github_id")
-    def picture_url = column[String]("picture_url")
-    def github_url = column[String]("github_url")
-    def email = column[String]("email")
-    def * = (id, login, name, github_id, picture_url, github_url, email) <> (User.tupled, User.unapply)
-  }
+object UserDoobieStore extends UserStore {
+  def all: ConnectionIO[List[User]] =
+    Queries.all.list
 
-  val users = TableQuery[Users]
+  def getByLogin(login: String): ConnectionIO[Option[User]] =
+    Queries.byLogin(login).option
 
-  override def all: Future[Seq[User]] = {
-    db.run(users.sortBy(_.id.desc).result)
-  }
+  def getById(id: Long): ConnectionIO[Option[User]] =
+    Queries.byId(id).option
 
-  override def getByLogin(login: String): Future[Seq[User]] = db.run(users.filter(_.login === login).result)
-
-  override def create(
-    login:       String,
-    name:        String,
-    github_id:   String,
-    picture_url: String,
-    github_url:  String,
-    email:       String
-  ): Future[User] = {
-
-    db.run {
-      (users returning users.map(_.id) into ((user, id) ⇒ user.copy(id = id))) += User(
-        id = None,
-        login = login,
-        name = name,
-        github_id = github_id,
-        picture_url = picture_url,
-        github_url = github_url,
-        email = email
-      )
-    }
-  }
+  def create(user: UserCreation.Request): ConnectionIO[UserCreation.Response] = for {
+    _ ← Queries.insert(user).run
+    user ← getByLogin(user.login)
+  } yield if (user.isDefined) Xor.Right(user.get) else Xor.Left(UserCreation.DuplicateName)
```

Comment:
A `fold` or `pattern matching` is better suited for this case than manual runtime boolean check on presence / absence. It's literally safe the way it is but we don't want to encourage use of `Option.get` or any unsafe state accessors anywhere